src/protocol.rs

```diff
@@ -481,7 +484,24 @@ mod tests {
             Err(panic) => {
                 if let Some(s) = panic.downcast_ref::<String>() {
                     println!("Panic occurred: {s}");
-                    assert_eq!(s, "Participant Config Size Error");
+                    assert_eq!(s, "Received a message with too few parties (Participant Config should have at least 2 parties)");
+                }
+            }
+        }
+
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
```

Comment:
note: you shouldn't need to create a new RNG halfway through the test, just use the one you had before.

Reply:
nit: RNG creation should happen outside the loop, not inside.